src/util/file.c

```diff
@@ -358,6 +358,30 @@ int fileput(char *path, char *string) {
     return(0);
 }
 
+int fileputat(int fd, char *path, char *string) {
```

Comment:
I would suggest we integrate this into 2.3 and use the new key/value registry to store this. So fileputat() will always just work.... Hrmm... Maybe the registry should store `void *` so it can take any sort of value (e.g. an int). I'll look into that too.